Source/WebCore/dom/Element.cpp

```diff
@@ -2498,6 +2499,52 @@ void Element::didMoveToNewDocument(Document& oldDocument, Document& newDocument)
             }
         }
     }
+
+    if (langAttrMatchesDocument()) {
+        oldDocument.removeElementWithLangAttrMatchingDocument(*this);
+        setLangAttrMatchesDocument(false);
+    }
+
+    setEffectiveLang();
+}
+
+void Element::setEffectiveLangFromParent()
+{
+    ASSERT(langFromAttribute().isNull());
+
+    if (auto* parent = parentOrShadowHostElement(); parent && parent != document().documentElement()) {
+        setEffectiveLangMatchesDocument(parent->effectiveLangMatchesDocument());
+        if (UNLIKELY(parent->hasRareData()) && !parent->elementRareData()->effectiveLang().isNull())
+            ensureElementRareData().setEffectiveLang(parent->elementRareData()->effectiveLang());
+        else if (hasRareData())
+            elementRareData()->setEffectiveLang(nullAtom());
+    }
+}
+
+void Element::setEffectiveLang()
```

Comment:
I'd call this updateEffectiveLangStates or updateEffectiveLangBitFlags.
We're not actually updating the effective lang. That happens in setEffectiveLangInSubtree.